index.js

```diff
@@ -50,27 +50,48 @@
 			function cb(e, r) {
 				err = e;
 				res = r;
-				done = true;		
+				done = true;
 			}
 		}
 	}
-	
+
 	module.exports = deasync;
-	
+	module.exports.default = deasync;
+
 	module.exports.sleep = deasync(function(timeout, done) {
 		setTimeout(done, timeout);
 	});
-	
+
 	module.exports.runLoopOnce = function(){
 		process._tickCallback();
 		binding.run();
 	};
-	
+
 	module.exports.loopWhile = function(pred){
 	  while(pred()){
 		process._tickCallback();
 		if(pred()) binding.run();
 	  }
 	};
 
+	module.exports.await = function(pr) {
+		var done, result;
+
+		done = false;
+		result = undefined;
+
+		pr
+			.then(function(r) {
+				done = true;
+				return result = r;
+			})
+			.catch(function(err) {
+				done = true
+				throw err
+			})
+
+		deasync.loopWhile(() => { return !done });
+		return result;
```

Comment:
```suggestion
		done = false;
		result = undefined;
		error = undefined;

		pr
			.then(function(r) {
				done = true;
				result = r;
			})
			.catch(function(err) {
				done = true;
				error = err;
			})

		deasync.loopWhile(function() { return !done; });

		if (error) throw error;
		return result;
```

Reply:
What about errors what are falsy? for example `Promise.reject(0)` or `Promise.reject(undefined)`.
This still would no throw an exception

Reply:
@NotWearingPants I forgot that those rejections are technically possible. Indeed to cover that it would be better to use a `didThrow` variable instead.